core/src/test/java/io/opencensus/stats/MeasureMapTest.java

```diff
@@ -0,0 +1,163 @@
+      builder.put(makeSimpleDoubleMeasure("m" + i), i * 11.1);
+      assertEquals(expected.iterator(), builder.build().iterator());
+    }
+  }
+
+  @Test
+  public void testDuplicateMeasures() {
+    // DoubleMeasure
+    MeasureMap metrics1 = MeasureMap.builder().put(M1, 1.0).build();
+    MeasureMap metrics2 = MeasureMap.builder().put(M1, 1.0).put(M2, 2.0).build();
+    assertEquals(MeasureMap.builder().put(M1, 1.0).put(M1, 2.0).build().iterator(),
+        metrics1.iterator());
+    assertEquals(MeasureMap.builder().put(M1, 1.0).put(M1, 2.0).put(M1, 3.0).build().iterator(),
+        metrics1.iterator());
+    assertEquals(MeasureMap.builder().put(M1, 1.0).put(M2, 2.0).put(M1, 3.0).build().iterator(),
+        metrics2.iterator());
+    assertEquals(MeasureMap.builder().put(M1, 1.0).put(M1, 2.0).put(M2, 2.0).build().iterator(),
+        metrics2.iterator());
+
+    // LongMeasure
+    MeasureMap metrics3 = MeasureMap.builder().put(M3, 100L).build();
+    MeasureMap metrics4 = MeasureMap.builder().put(M3, 100L).put(M4, 200L).build();
+    assertEquals(MeasureMap.builder().put(M3, 100L).put(M3, 100L).build().iterator(),
+        metrics3.iterator());
+    assertEquals(MeasureMap.builder().put(M3, 100L).put(M3, 200L).put(M3, 300L).build().iterator(),
+        metrics3.iterator());
+    assertEquals(MeasureMap.builder().put(M3, 100L).put(M4, 200L).put(M3, 300L).build().iterator(),
+        metrics4.iterator());
+    assertEquals(MeasureMap.builder().put(M3, 100L).put(M3, 200L).put(M4, 200L).build().iterator(),
+        metrics4.iterator());
+
+    // Mixed
+    MeasureMap metrics5 = MeasureMap.builder().put(M3, 100L).put(M1, 1.0).build();
+    MeasureMap metrics6 = MeasureMap.builder().put(M2, 2.0).put(M3, 200L).build();
+    assertEquals(MeasureMap.builder().put(M3, 100L).put(M1, 1.0).put(M3, 300L).build().iterator(),
+        metrics5.iterator());
+    assertEquals(MeasureMap.builder().put(M2, 2.0).put(M3, 200L).put(M2, 3.0).build().iterator(),
+        metrics6.iterator());
+  }
+
+  private static final DoubleMeasure M1 = makeSimpleDoubleMeasure("m1");
+  private static final DoubleMeasure M2 = makeSimpleDoubleMeasure("m2");
+  private static final LongMeasure M3 = makeSimpleLongMeasure("m3");
+  private static final LongMeasure M4 = makeSimpleLongMeasure("m4");
+
+  private static final DoubleMeasure makeSimpleDoubleMeasure(String measure) {
+    return Measure.DoubleMeasure.create(
+        measure, measure + " description", "1");
+  }
+
+  private static final LongMeasure makeSimpleLongMeasure(String measure) {
+    return Measure.LongMeasure.create(
+        measure, measure + " description", "1");
+  }
+
+  private static void assertEquals(
+      Iterator<Measurement> expected, Iterator<Measurement> actual) {
+    while (expected.hasNext() && actual.hasNext()) {
+      Measurement expectedMeasurement = expected.next();
+      Measurement actualMeasurement = actual.next();
```

Comment:
I don't think we should compare the element order, since MeasureMap could be implemented with a map. (`containsExactly` would also avoid testing the order.)